File: leetcode/no1512_Number_of_Good_Pairs.go
package main

import "fmt"

func numIdenticalPairs(nums []int) int {

	counter := 0

	for i, numA := range nums {

		for j, numB := range nums {

			if j > i {
				if numA == numB {
					counter += 1
				}
			}
		}
	}

	return counter
}

func main() {
	nums := []int{1, 2, 3, 1, 1, 3}
	ans := numIdenticalPairs(nums)
	fmt.Println("OUTPUT:", ans)
}

File: leetcode/no1512_Number_of_Good_Pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
    let result = [];

    for (let i = 0; i < nums.length; i++) {

        for (let j = i; j < nums.length; j++) {

            if (i == j) {
                continue;
            }
            if (nums[i] == nums[j]) {
                result.push([i, j]);
            }
        }
    }

    return result.length;
};
